### PR TITLE
Template supermarket json

### PR DESCRIPTION
**Pull request by stephenlauck**
I would like to manage the supermarket.json with a template instead of writing it out as a file. I am also not sure it's a good idea to do checking on the values and fail but better to just leave it to the user to make sure they populate the correct values.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -49,3 +31,12 @@
...
 end
+
+template '/etc/supermarket/supermarket.json' do
+  source 'supermarket.json.erb'
+  owner 'root'
+  group 'root'
+  mode "0644"
+  action :create
+  notifies :reconfigure, 'chef_server_ingredient[supermarket]'
+end
```
Comment (irvingpop) [inline recipes/default.rb]: could you please add newline to the EOF and then I'm :+1:

**Comment by david-crowder** [comment]
The current implementation seems to allow you to place any attribute you want in the supermarket.json (for instance if you want to disable ['nginx']['force_ssl'] :smiley:), but the new template only supports a handful of config.

**Comment by irvingpop** [comment]
@david-crowder your sentiments are why I'm hesitant to merge this.  I also like the flexibility of the JSON output.

@stephenlauck I'm worried that we'd be breaking a number of existing user's use cases like David's with this PR.   How do you think we should proceed?

**Comment by stephenlauck** [comment]
I agree with the above and I think it's a good idea to just take whatever hash is assembled from the cookbook attributes/recipes and just drop the key value pairs into the template. I'll try to get to this or if @david-crowder or @irvingpop want to pair up and fix this let me know.

**Comment by irvingpop** [comment]
@stephenlauck what's the advantage of doing that with a template vs just writing out the attributes to JSON like it's doing now?

**Comment by stephenlauck** [comment]
I was thinking that we may end up with a more complicated template like chef-server https://github.com/chef-cookbooks/chef-server/blob/master/templates/default/chef-server.rb.erb

**Comment by irvingpop** [comment]
I still don't see how that's better :)

**Pull request closed without merging**